app/src/main/java/com/ubergeek42/WeechatAndroid/adapters/ChatLinesAdapter.kt

```diff
@@ -332,10 +336,24 @@ class ChatLinesAdapter @MainThread constructor(
         }
 
         override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
-            if (newItemPosition == 0) {
-                return oldLines[oldItemPosition] === newLines[newItemPosition]
+            return if (newItemPosition == 0) {
+                // Header line, always present
+                oldLines[oldItemPosition] === newLines[newItemPosition]
+            } else if (!sameStyle) {
+                false
+            } else if (diffLineContents) {
+                val oldLine = oldLines[oldItemPosition]
+                val newLine = newLines[newItemPosition]
+
+                oldLine.type == newLine.type &&
+                oldLine.timestamp == newLine.timestamp &&
+                oldLine.rawPrefix == newLine.rawPrefix &&
+                oldLine.rawMessage == newLine.rawMessage &&
+                oldLine.isHighlighted == newLine.isHighlighted &&
+                oldLine.displayAs == newLine.displayAs
```

Comment:
Can this equality / "key" crap be a getter of the Line type? The repetition hurts my eyes.

Reply:
Created a `visuallyEqualsTo` in `Line`. Not a fan of this all but should be good enough